jax_rocm_plugin/build/rocm/tools/get_rocm.py

```diff
@@ -144,32 +160,29 @@ def get_system():
         if md["PLATFORM_ID"] == "platform:el8":
             return RHEL8
 
-    raise Exception("No system for %r" % md)
+    raise ROCmInstallException("No system for %r" % md)
 
 
 def _setup_internal_repo(system, rocm_version, job_name, build_num):
+    """Set up repos for getting internal ROCm builds"""
     # wget is required by amdgpu-repo
     system.install_packages(["wget"])
 
     install_amdgpu_installer_internal(rocm_version)
 
-    amdgpu_build = (
-        urllib.request.urlopen(
-            "http://rocm-ci.amd.com/job/%s/%s/artifact/amdgpu_kernel_info.txt"
-            % (job_name, build_num)
-        )
-        .read()
-        .decode("utf8")
-        .strip()
-    )
+    with urllib.request.urlopen(
+        "http://rocm-ci.amd.com/job/%s/%s/artifact/amdgpu_kernel_info.txt"
+        % (job_name, build_num)
+    ) as kernel_info:
+        amdgpu_build = kernel_info.read().decode("utf8").strip()
```

Comment:
I'm not sure this is going to be functionally the same as just reading the full request without the with block. If anything just put it in a try-finally and close the socket or whatever pylint is complaining about.

I really don't think we should put the subprocess call into the same block as the URL thing.

Reply:
Okay, then since they're functionally equivalent, we can just use the context manager and make pylint happy :)